engine/table/src/main/java/io/deephaven/engine/table/impl/MultiJoinTableImpl.java

```diff
@@ -466,21 +522,21 @@ protected void process() {
                 downstream.added = RowSetFactory.empty();
             }
 
-            final long[] currentRedirections = new long[stateManager.getTableCount()];
-            final boolean[] modifiedTables = new boolean[stateManager.getTableCount()];
-            final boolean[] modifiedTablesOnThisRow = new boolean[stateManager.getTableCount()];
+            final long[] currentRedirections = new long[tableCount];
+            final boolean[] modifiedTables = new boolean[tableCount];
+            final boolean[] modifiedTablesOnThisRow = new boolean[tableCount];
 
-            final RowSetBuilderRandom modifiedBuilder = RowSetFactory.builderRandom();
-            final RowSetBuilderRandom removedBuilder = RowSetFactory.builderRandom();
-            final RowSetBuilderRandom addedBuilder = RowSetFactory.builderRandom();
+            final RowSetBuilderRandom modifiedBuilder = new BitmapRandomBuilder((int) newSize);
+            final RowSetBuilderRandom emptiedBuilder = new BitmapRandomBuilder((int) newSize);
+            final RowSetBuilderRandom reincarnatedBuilder = new BitmapRandomBuilder((int) newSize);
```

Comment:
Maybe use regular random builders for these? Might want to run by Charles as a straw-man.

Reply:
Agreed to use BMBuilder for modifications, Random for emptied/reincarnated similar to Agg usage.